### PR TITLE
[HWKMETRICS-13]Adding InfluxDB java client support test

### PR DESCRIPTION
**Pull request by jkandasa**

### CODE REVIEW

**Comment by rhqci** [comment]
Can one of the admins verify this patch?

**Comment by tsegismont** [comment]
I'll have a look tomorrow morning. Thanks


-----Original Message-----

**Comment by jkandasa** [comment]
@tsegismont  Hi Thomas, There are two tests failed due to incorrect value from hawkular-metrics server.
1.Test Sum
2.Test Mean
Input:10.9,11.9,12.9,13.9,14.9,15.9,16.9,17.9,18.9
Expected Sum:134.1
Expected Mean:14.9
Result from hawkular-metrics:
Sum:134.10000000000002
Mean:14.900000000000002

**Comment by tsegismont** [comment]
You can't use assertEquals with double values. IIRC, junit has an assertDoubleEquals method which takes an extra delta argument.


-----Original Message-----

File: rest-tests/src/test/java/org/hawkular/metrics/test/InfluxDatabaseITest.java [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+        Serie serieSource = new Serie.Builder(timeSeries)
+        .columns("time", "sum")
+        .values(new Double(startTime+8000L), 134.1D)
+        .build();
+        List<Serie> series = influxDB.query(dbName,
+                                            "select sum(value) from "+timeSeries
+                                            +" where time > now() - 30s group by time(30s)",
+                                            TimeUnit.MILLISECONDS);
+        Assert.assertNotNull(series);
+        Assert.assertEquals(serieSource.toString(), series.get(0).toString());
```
Comment (tsegismont) [inline rest-tests/src/test/java/org/hawkular/metrics/test/InfluxDatabaseITest.java]: Ok now I see the issue. You can't compare the two series as string. You may create an `#assertSeriesEquals` method, in this class or its parent. In this method you could iterate over the points and check that the timestamps and values are "equal" within a delta.

See http://junit.org/javadoc/latest/org/junit/Assert.html#assertEquals%28java.lang.String,%20double,%20double,%20double%29

**Comment by tsegismont** [comment]
This is a very good contribution @jkandasa , thanks. Waiting for the test failure fix before merging.

**Comment by jkandasa** [comment]
@tsegismont Thank you Thomas. I have changed 'assertDoubleEquals' for double values with delta as '1e-13' for double and '1e-7' for float

**Comment by tsegismont** [comment]
@jkandasa perfect! Merging now.

**Comment by pilhuhn** [comment]
(Just a more general thought, not related to this PR, but triggered by it)  
This double compare failure (and other stuff) really should make us think if we really want to treat every data value as double instead of integral types for integral values.

**Pull request merged**